apps/api/src/app/widgets/pipes/limit-pipe/limit-pipe.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { expect } from 'chai';
+import { LimitPipe } from './limit-pipe';
+import { Paramtype } from '@nestjs/common/interfaces/features/paramtype.interface';
+
+enum METADATA {
+  DATA = 'limit',
+  TYPE = 'query',
+}
+
+describe('LimitPipe', () => {
+  let pipe: LimitPipe;
+  const metadata = { data: METADATA.DATA, type: METADATA.TYPE as Paramtype, metatype: String };
+
+  beforeEach(() => {
+    pipe = new LimitPipe(1, 1000);
+  });
+
+  it('should return the input value if it is within the limits', () => {
+    let limit = 500;
+    let res = pipe.transform(limit, metadata);
+    expect(res).to.equal(limit);
+
+    limit = -1;
+    expect(() => pipe.transform(limit, metadata)).to.throw(`${METADATA.DATA} must not be less than 1`);
+
+    limit = 0;
+    expect(() => pipe.transform(limit, metadata)).to.throw(`${METADATA.DATA} must not be less than 1`);
+
+    limit = 1;
+    res = pipe.transform(limit, metadata);
+    expect(res).to.equal(limit);
+
+    limit = 500;
+    res = pipe.transform(limit, metadata);
+    expect(res).to.equal(limit);
+
+    limit = 999;
+    res = pipe.transform(limit, metadata);
+    expect(res).to.equal(limit);
+
+    limit = 1000;
+    res = pipe.transform(limit, metadata);
+    expect(res).to.equal(limit);
+
+    limit = 1001;
+    expect(() => pipe.transform(limit, metadata)).to.throw(`${METADATA.DATA} must not be greater than 1000`);
+  });
```

Comment:
I think its better to split this test into smaller tests, like with the `it.each`` `, so when a few cases will fail you would know which tests to fix... same about the below code

Reply:
Yeah, I was not sure here is to split those or not. Do you think split ones will be easier to read?

Reply:
like easier to spot the failing parts when you will run the test... so like if you will have let's say 10 tests, you will immediately see that for example three are failing and you will know which are these three tests. 
If you would have one test you would have to run it many times, like run, spot the issue, fix the issue, then run again, spot the issue, fix etc.